hooks/useCreateData.tsx

```diff
@@ -33,10 +37,21 @@ const useCreateData = () => {
     dummyRandom,
   } = useCharacter()
   const { purchaseWithComment } = useZoraMintByPrivy()
+  const { mint: purchasePresaleWithComment } = useZoraPremint()
   const { setCurrentStatus } = useAnimatedBook()
   const { renderSinglePfp, renderMultiplePfps } = usePfpRenderer()
   const { renderSingleSheet, renderMultipleSheets } = useSheetRenderer()
   const { prepare } = usePreparePrivyWallet()
+  const { externalWallet } = useConnectedWallet()
+
+  useEffect(() => {
+    const checkIfWhitelisted = () => {
+      const lowercaseAddress = externalWallet?.address
+      const isWhitelistedAddress = whitelistedAddresses.includes(lowercaseAddress)
+      setIsWhitelisted(isWhitelistedAddress)
+    }
+    checkIfWhitelisted()
+  }, [externalWallet])
```

Comment:
can you move this logic to a new hook?
- `hooks/useWhitelist`
- returns the `isWhitelisted` variable for easy import into any part of the application